Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Spawning/CreateSpawnTicketNodeable.h

```diff
@@ -8,19 +8,26 @@
 
 #pragma once
 
+#include <AzFramework/Spawnable/SpawnableAssetRef.h>
 #include <AzFramework/Spawnable/SpawnableEntitiesInterface.h>
+#include <AzFramework/Spawnable/SpawnableMediator.h>
+#include <Include/ScriptCanvas/Libraries/Spawning/CreateSpawnTicketNodeable.generated.h>
 #include <ScriptCanvas/CodeGen/NodeableCodegen.h>
 #include <ScriptCanvas/Core/Node.h>
 #include <ScriptCanvas/Core/Nodeable.h>
-#include <Libraries/Spawning/SpawnableAsset.h>
-#include <Libraries/Spawning/SpawnTicketInstance.h>
-#include <Include/ScriptCanvas/Libraries/Spawning/CreateSpawnTicketNodeable.generated.h>
 
 namespace ScriptCanvas::Nodeables::Spawning
 {
     class CreateSpawnTicketNodeable
-        : public ScriptCanvas::Nodeable
+        : public Nodeable
     {
         SCRIPTCANVAS_NODE(CreateSpawnTicketNodeable);
+    public:
+        CreateSpawnTicketNodeable() = default;
+        CreateSpawnTicketNodeable(const CreateSpawnTicketNodeable& rhs);
+        CreateSpawnTicketNodeable& operator=(const CreateSpawnTicketNodeable& rhs);
+
+    private:
+        AzFramework::SpawnableMediator m_spawnableMediator;
```

Comment:
Why does this node require s mediator. All this node is doing is creating and returning an EntitySpawnTicket. It can be done without instantiating a mediator as well

Reply:
I just wanted to everything go through a single system. I am ok either way if you think it's better.

Reply:
If I understand the role of the mediator correctly, it exists to collect the entities from spawn/despawn requests and trigger appropriate notifications with them. Creation of a ticket happens instantaneously and doesn't need a mediator. We will be allocating memory we won't be using. So I think this shouldn't be needed.

Reply:
it's more of providing an interface for Lua/SC for calling upon SpawnableAPI. e.g. in Lua we can't directly create EntitySpawnTicket because spawnableAsset is not reflectable, so instead we have to rely on mediator to do this using SpawnableAssetRef type which can be reflected with BC.